Can't build centos package because request to mirror site returns 403 Forbidden error

Following the build instructions for Linux results in a dead end when trying to build a package for centos. In the centos Vagrant box,

```
cd /vagrant
make deps

result:
[20 min or so of successful build execution]
--2015-02-03 15:52:09--  http://mirror.anl.gov/pub/gnu/libtool/libtool-2.4.2.tar.gz
Resolving mirror.anl.gov... 146.137.96.7, 2620:0:dc0:1800:214:4fff:fe7d:1b9
Connecting to mirror.anl.gov|146.137.96.7|:80... connected.
HTTP request sent, awaiting response... 403 Forbidden
2015-02-03 15:52:09 ERROR 403: Forbidden.

make: *** [deps] Error 8
```

I got around this by running `sudo yum install lilbtool` before `make deps`

But now it fails because it can't find the proper version of autoconf

```
/vagrant/.sources/thrift-0.9.1 /vagrant/.sources
configure.ac:20: error: Autoconf version 2.65 or higher is required
configure.ac:20: the top level
autom4te: /usr/bin/m4 failed with exit status: 63
autoscan: /usr/bin/autom4te failed with exit status: 63
make: *** [deps] Error 63
```

I tried the master branch and branch new-build-deux. I also tried changing Vagrantfile to build from chef/centos-6.6 instead of chef/centos-6.5. The same behavior above occurred in all those scenarios.

autoconf-2.63-5.1.el6.noarch is installed, and yum search reveals only autoconf213.noarch. The epel repo is installed, but I don't know where I'm supposed to get the required version of autoconf. Did I manage to do something wrong in these ultra easy instructions, or is the build config broken for centos?

I also tried using chef/centos-7.0 as the base VM. Then when I run 'make deps' I get a different failure:

http://people.centos.org/tru/devtools-2/7/x86_64/RPMS/repodata/repomd.xml [Errno 14] HTTP Error 404 - Not Found

The yum repo at that address exists, but there is no support for version 7, only 5 and 6; hence the 404 error.

The ubuntu14 VM also doesn't build

```
CMake Error: Generator: execution of make failed. Make command was: /usr/bin/gmake "cmTryCompileExec686272267/fast"
-- Check for working CXX compiler: /usr/bin/clang++ -- broken
CMake Error at /usr/share/cmake-2.8/Modules/CMakeTestCXXCompiler.cmake:54 (message):
  The C++ compiler "/usr/bin/clang++" is not able to compile a simple test
  program.

  It fails with the following output:

   Change Dir: /vagrant/.sources/gflags-2.1.1/CMakeFiles/CMakeTmp

  

  Run Build Command:/usr/bin/gmake "cmTryCompileExec686272267/fast"

  No such file or directory

  Generator: execution of make failed.  Make command was: /usr/bin/gmake
  "cmTryCompileExec686272267/fast"
```

When I try that, it can't find cmake

```$ vagrant ssh ubuntu14
Welcome to Ubuntu 14.04.1 LTS (GNU/Linux 3.13.0-45-generic x86_64)

 * Documentation:  https://help.ubuntu.com/

  System information as of Tue Feb  3 19:39:09 UTC 2015

  System load:  0.03              Processes:           85
  Usage of /:   2.8% of 39.34GB   Users logged in:     0
  Memory usage: 3%                IP address for eth0: 10.0.2.15
  Swap usage:   0%

  Graph this data and manage this system at:
    https://landscape.canonical.com/

  Get cloud support with Ubuntu Advantage Cloud Guest:
    http://www.ubuntu.com/business/services/cloud

0 packages can be updated.
0 updates are security updates.


Last login: Tue Feb  3 19:39:09 2015 from 10.0.2.2
vagrant@vagrant-ubuntu-trusty-64:~$ sudo su && /vagrant
root@vagrant-ubuntu-trusty-64:/home/vagrant# cd /vagrant
root@vagrant-ubuntu-trusty-64:/vagrant# make -j2
mkdir -p build/trusty
ln -snf trusty build/linux
cd build/trusty && cmake ../.. && \
		make --no-print-directory  --jobserver-fds=3,4 -j
/bin/sh: 1: cmake: not found
make: *** [all] Error 127
root@vagrant-ubuntu-trusty-64:/vagrant# 
``` 
Note that sudo su && cd /vagrant actually puts me in /home/vagrant where there is no Makefile, so added dc /vagrant.

I installed cmake from apt-get, and then this happens

```
root@vagrant-ubuntu-trusty-64:/vagrant# make -j2
mkdir -p build/trusty
ln -snf trusty build/linux
cd build/trusty && cmake ../.. && \
		make --no-print-directory  --jobserver-fds=3,4 -j
-- The C compiler identification is unknown
-- The CXX compiler identification is unknown
-- Check for working C compiler: /usr/bin/clang
CMake Error: your C compiler: "/usr/bin/clang" was not found.   Please set CMAKE_C_COMPILER to a valid compiler path or name.
CMake Error: Internal CMake error, TryCompile configure of cmake failed
-- Check for working C compiler: /usr/bin/clang -- broken
CMake Error at /usr/share/cmake-2.8/Modules/CMakeTestCCompiler.cmake:61 (message):
  The C compiler "/usr/bin/clang" is not able to compile a simple test
  program.

  It fails with the following output:

   

  

  CMake will not be able to correctly generate this project.
Call Stack (most recent call first):
  CMakeLists.txt:48 (project)


CMake Error: your C compiler: "/usr/bin/clang" was not found.   Please set CMAKE_C_COMPILER to a valid compiler path or name.
CMake Error: your CXX compiler: "/usr/bin/clang++" was not found.   Please set CMAKE_CXX_COMPILER to a valid compiler path or name.
-- Configuring incomplete, errors occurred!
See also "/vagrant/build/trusty/CMakeFiles/CMakeOutput.log".
See also "/vagrant/build/trusty/CMakeFiles/CMakeError.log".
make: *** [all] Error 1
```

--- a/tools/provision.sh
+++ b/tools/provision.sh
@@ -184,15 +184,15 @@ function install_automake() {
 
 function install_libtool() {
   if [[ ! -f /usr/bin/libtool ]]; then
-    if [[ ! -f libtool-2.4.2.tar.gz ]]; then
-      wget http://mirror.anl.gov/pub/gnu/libtool/libtool-2.4.2.tar.gz
+    if [[ ! -f libtool-2.4.5.tar.gz ]]; then
+      wget http://ftpmirror.gnu.org/libtool/libtool-2.4.5.tar.gz
     else
       log "libtool is already downloaded. skipping."
     fi
-    if [[ ! -d libtool-2.4.2 ]]; then
-      tar -xf libtool-2.4.2.tar.gz
+    if [[ ! -d libtool-2.4.5 ]]; then
+      tar -xf libtool-2.4.5.tar.gz
     fi
-    pushd libtool-2.4.2
+    pushd libtool-2.4.5
     ./configure --prefix=/usr
     make
     sudo make install

